java-strings/src/test/java/com/baeldung/string/StringFromPrimitiveArrayUnitTest.java

```diff
@@ -0,0 +1,96 @@
+package com.baeldung.string;
+
+import com.google.common.base.Joiner;
+import com.google.common.primitives.Chars;
+import com.google.common.primitives.Ints;
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.junit.Test;
+
+import java.nio.CharBuffer;
+import java.util.Arrays;
+import java.util.StringJoiner;
+import java.util.stream.Collectors;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+
+public class StringFromPrimitiveArrayUnitTest {
+
+    private int[] intArray = {1, 2, 3, 4, 5, 6, 7, 8, 9};
+
+    private char[] charArray = {'a', 'b', 'c', 'd', 'e', 'f'};
+
+    private char separatorChar = '-';
+
+    private String separator = String.valueOf(separatorChar);
+
+    private String expectedIntString = "1-2-3-4-5-6-7-8-9";
+
+    private String expectedCharString = "a-b-c-d-e-f";
+
+    @Test
+    public void givenPrimitiveArray_whenJoinBySeparator_thenReturnsString_through_Java8CollectorsJoining() {
```

Comment:
Let's have separate test methods for the int array vs the char array for all the examples. The test names could be like givenIntArray_... and givenCharArray_...

Reply:
Ok, I'll split each test into two